### PR TITLE
Add tokenFunc argument to `WorkspaceDb.getWorkspaceDb`

### PR DESCRIPTION
**Pull request by kabentley**
This is a breaking change to an @beta api to provide a way to supply the `AccessToken` for a container found via a dbAlias. The method itself is now async.

### CODE REVIEW

File: core/backend/src/workspace/Workspace.ts [hunk 1]
```diff
@@ -70,6 +70,16 @@ export namespace WorkspaceContainer {
     /** attempt to synchronize (i.e. call `checkForChanges`) this cloud container whenever it is connected to a cloud cache. */
     syncOnConnect?: boolean;
   }
+
+  /** A function to supply an [AccessToken]($bentley) for a `WorkspaceContainer`.
+   * @returns a Promise that resolves to the AccessToken for the container.
+   */
+  export type TokenFunc = (
+    /** The properties of the WorkspaceContainer necessary to obtain the access token */
```
Comment (pmconne) [inline core/backend/src/workspace/Workspace.ts:78]: Will these doc comments on `props` and `accounts` actually show up in the docs?

File: core/backend/src/workspace/Workspace.ts [hunk 2]
```diff
@@ -244,7 +256,7 @@ export interface Workspace {
    * @param databaseName the database name, resolved via [[resolveDatabase]].
    * @see [[getWorkspaceDbFromProps]]
```
Comment (pmconne) [inline core/backend/src/workspace/Workspace.ts:257]: new param not documented.

File: core/backend/src/workspace/Workspace.ts (hunk 1 above)
Comment (kabentley) [reply core/backend/src/workspace/Workspace.ts:78]: i think so, i'll have to check.

Comment (kabentley) [reply core/backend/src/workspace/Workspace.ts]: reworked the "conventional" way

File: core/backend/src/workspace/Workspace.ts (hunk 2 above)
Comment (kabentley) [reply core/backend/src/workspace/Workspace.ts]: done

**Review by pmconne — APPROVED** [review]

**Pull request merged**